study-builder/tenants/register-user-in-pepper.js

```diff
@@ -136,6 +136,20 @@ function (user, context, callback) {
             console.log('No Registration Mode passed in request');
         }
 
+        if (pepper_params.mode && pepper_params.mode === "login" &&
+            !user.app_metadata.user_guid &&
+            !pepper_params.tempUserGuid
+        ) {
+            console.log('User not registered');
+            const loginErrPayload = {
+                code: 'user_not_registered',
+                message: 'User need to register first in order to login',
+                statusCode: 422,
```

Comment:
A `403 Forbidden` would fit better here than a `422 Unprocessable Entity`. There's nothing wrong with the request, the user is just not permitted to access the resource- regardless of their authentication state.